hydradx/model/amm/basilisk_amm.py

```diff
@@ -0,0 +1,152 @@
+import copy
+import math
+
+
+class BasiliskPoolState:
```

Comment:
That is true. Which puts the question in my mind, what exactly is unique about Basilisk, and at what point does that come in? I had been thinking there would be only one more layer, which brings together agents, pools and the external market, and serves as the access point for all state change functions (basically, the API layer). But possibly there needs to be another layer that collects all the constant product pools into a complete (Basilisk) exchange?

Reply:
unique about basilisk compared to other AMM projects that are just collections of constant product AMMs? Not much yet. We will introduce some new things at the chain level (instead of at the AMM/math level) but not sure when we will get around to modeling them. So yeah, for now we're really just implementing a collection of constant product AMMs... could be Uniswap v2, or Basilisk, or Sushiswap, etc.

Reply:
I feel like there should be something named Basilisk somewhere...

Reply:
I don't think it needs to until we incorporate something that is specific to Basilisk, and there isn't much special about Basilisk AMMs yet. Maybe once we have a router...